# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1544734395 -3600
#      Thu Dec 13 21:53:15 2018 +0100
# Node ID 3a90eff58fb5d7d85c78f8f9cda8163c60814994
# Parent  1a12d4369beb7244e4bfa641493ebb8c12cf6edf
editor encodings in prefs dialog only in  uppercase (bug #55139)

* resource-manager.cc (do_combo_encoding): convert name of codec for locale into
  upper case, do not search entries in combo box case sensitive

diff --git a/libgui/src/resource-manager.cc b/libgui/src/resource-manager.cc
--- a/libgui/src/resource-manager.cc
+++ b/libgui/src/resource-manager.cc
@@ -407,33 +407,33 @@ namespace octave
       {
         enc = m_settings->value (ed_default_enc.key, ed_default_enc.def).toString ();
 
         if (enc.isEmpty ())  // still empty?
           {
             if (default_exists)
               enc = ed_default_enc.def.toString ();
             else
-              enc = QTextCodec::codecForLocale ()->name ();
+              enc = QTextCodec::codecForLocale ()->name ().toUpper ();
           }
       }
 
     // fill the combo box
     foreach (QString c, all_codecs)
       combo->addItem (c);
 
     // prepend the default item
     combo->insertSeparator (0);
     if (default_exists)
       combo->insertItem (0, ed_default_enc.def.toString ());
     else
-      combo->insertItem (0, QTextCodec::codecForLocale ()->name ());
+      combo->insertItem (0, QTextCodec::codecForLocale ()->name ().toUpper ());
 
     // select the default or the current one
-    int idx = combo->findText (enc);
+    int idx = combo->findText (enc, Qt::MatchExactly);
     if (idx >= 0)
       combo->setCurrentIndex (idx);
     else
       combo->setCurrentIndex (0);
 
     combo->setMaxVisibleItems (12);
   }
 }
